JitsiConference.js

```diff
@@ -2077,11 +2081,75 @@ JitsiConference.prototype.sendEndpointMessage = function(to, payload) {
  * Sends a broadcast message via the data channel.
  * @param payload {object} the payload of the message.
  * @throws NetworkError or InvalidStateError or Error if the operation fails.
+ * @deprecated Use 'sendMessage' instead. TODO: this should be private.
  */
 JitsiConference.prototype.broadcastEndpointMessage = function(payload) {
     this.sendEndpointMessage('', payload);
 };
 
+/**
+ * Sends a message to a given endpoint (if 'to' is a non-empty string), or
+ * broadcasts it to all endpoints in the conference.
+ * @param {string} to The ID of the endpoint/participant which is to receive
+ * the message, or '' to broadcast the message to all endpoints in the
+ * conference.
+ * @param {string|object} message the message to send. If this is of type
+ * 'string' it will be sent as a chat message. If it is of type 'object', it
+ * will be encapsulated in a format recognized by jitsi-meet and converted to
+ * JSON before being sent.
+ * @param sendThroughVideobridge Whether to send the message through
+ * jitsi-videobridge (via the COLIBRI data channel or web socket), or through
+ * the XMPP MUC. Currently only objects can be sent through jitsi-videobridge.
+ */
+JitsiConference.prototype.sendMessage = function(
+        message,
+        to = '',
+        sendThroughVideobridge = false) {
+
+    if (sendThroughVideobridge) {
+        if (!(typeof message === 'object')) {
+            logger.error(`Can not send message of type ${
+                typeof message} through jitsi-videobridge.`);
+
+            return;
+        }
+        this.sendEndpointMessage(to, message);
+    } else {
+        // We only support strings (i.e. regular text messages) and objects (
+        // encapsulated in a specific JSON format)
+        if (typeof message !== 'string' && typeof message !== 'object') {
+            logger.error(`Can not send a message of type  ${typeof message}`);
+
+            return;
+        }
+
+        let messageToSend = message;
+
+        if (typeof message === 'object') {
+            // Encapsulate the object in the jitsi-meet format, and convert it
+            // to JSON.
+            messageToSend = { payload: message };
+            messageToSend[JITSI_MEET_MUC_TOPIC] = '';
+
+            try {
+                messageToSend = JSON.stringify(messageToSend);
+            } catch (e) {
+                logger.error('Can not send a message, stringify failed: ', e);
+
+                return;
+            }
+        }
+
+        if (to === '') {
+            // Broadcast
```

Comment:
This comment should probably be more explicit if it's needed at all. Saying "broadcast" my itself doesn't say much to someone like me because I could use the word (maybe incorrectly as) I broadcast/send my message to specific people.